CMakeLists.txt

```diff
@@ -1,303 +1,159 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
-PROJECT(ARRAYFIRE)
+cmake_minimum_required(VERSION 3.5.2)
```

Comment:
@9prady9 16.04 has 3.5.2. I think that should be OK for 3.6.0 and forward. Anyone else can install cmake from source.